panther_analysis_tool/backend/public_api_client.py

```diff
@@ -127,40 +140,40 @@ def check(self) -> BackendCheckResponse:
             success=True, message=f"connected to Panther backend on version: {panther_version}"
         )
 
+    def async_bulk_upload(self, params: BulkUploadParams) -> BackendResponse[BulkUploadResponse]:
+        query = self._requests.async_bulk_upload_mutation()
+        upload_params = {"input": {"data": params.encoded_bytes(), "patVersion": VERSION_STRING}}
+        res = self._safe_execute(query, variable_values=upload_params)
+        receipt_id = res.data.get("uploadDetectionEntitiesAsync", {}).get("receiptId")  # type: ignore
+        if not receipt_id:
+            raise BackendError("empty data")
+
+        while True:
+            query = self._requests.async_bulk_upload_status_query()
+            params = {"input": receipt_id}  # type: ignore
+            res = self._safe_execute(query, variable_values=params)  # type: ignore
+            result = res.data.get("detectionEntitiesUploadStatus", {})  # type: ignore
+            status = result.get("status", "")
+            error = result.get("error")
+            data = result.get("result")
+            if status == "FAILED":
+                if is_retryable_error_str(error):
+                    raise BackendError(error)
+                raise PermanentBackendError(error)
+            if status == "":
+                raise BackendError("empty data")
+
+            if status == "COMPLETED":
+                return to_bulk_upload_response(data)
+            time.sleep(5)
```

Comment:
Can we move this to the top of this loop so it sleeps before the first check?

Can we also change this to `2` maybe? `5` seems too long

Reply:
sure thing

Reply:
done